### PR TITLE
Code optimization for add additional information to log trace in api server

### LINKED ISSUE #96408: Code optimization for add additional information to log trace in api server
**What would you like to be added**:
The following code is not good for reading, we can provide a function to improve readability.
https://github.com/kubernetes/kubernetes/blob/4628c605aadb9b3bdf82b3d9e1a2d77580eefcaf/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go#L64

Such as, 
function getTraceFields(req *http.Request) []utiltrace.Field {
    return []utiltrace.Field{
           Key: "user-agent", Value: &lazyTruncatedUserAgent{req}}, 
           utiltrace.Field{Key: "client", Value: &lazyClientIP{req}
    }
}

Besides, we can add addtional information such request url, application lay protocol and so on.
**Why is this needed**:
To improve code readability and delimitation ability.

### PR DESCRIPTION
**Pull request by lala123912**
What type of PR is this?
/kind feature

What this PR does / why we need it:

Code optimization for log trace
add "Accept" and "protocol" to log trace
Which issue(s) this PR fixes:
Fixes #96408

Special notes for your reviewer:

Does this PR introduce a user-facing change?:
NONE

Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/trace_util.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+package handlers
+
+import (
+	"net/http"
+
+	utiltrace "k8s.io/utils/trace"
+)
+
+func getTraceFields(req *http.Request) []utiltrace.Field {
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/trace_util.go:25]: s/getT/t/

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 		// TODO: we either want to remove timeout or document it (if we document, move timeout out of this function and declare it in api_installer)
 		timeout := parseTimeout(req.URL.Query().Get("timeout"))
 
+
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go:63]: no double empty lines please

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/trace_util.go (hunk 1 above)
Comment (lala123912) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/trace_util.go:25]: @sttts  Could you give more detail information?

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/trace_util.go:25]: oh, just remove the `get` from the function name. It's not a getter.

**Comment by fedebongio** [comment]
/assign @sttts 
/triage accepted
Thank you Stefan for proactively looking at this one!

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/trace_util.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+	utiltrace "k8s.io/utils/trace"
+)
+
+func traceFields(req *http.Request) []utiltrace.Field {
+	return []utiltrace.Field{
+		{Key: "url", Value: req.URL.Path},
+		{Key: "user-agent", Value: &lazyTruncatedUserAgent{req: req}},
+		{Key: "client", Value: &lazyClientIP{req: req}},
+		{Key: "accept", Value: req.Header.Get("Accept")},
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/trace_util.go:30]: can we wrap this line into a lazy struct too? If I see this correctly, it does string normalization internally, similarly to lazyTruncatedUserAgent.

Comment (lala123912) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/trace_util.go]: It's a good idea, and I have fixed it in the new commit.

**Pull request merged**